engines/director/graphics.cpp

```diff
@@ -241,8 +241,8 @@ void inkDrawPixel(int x, int y, int src, void *data) {
 		byte rSrc, gSrc, bSrc;
 		byte rDst, gDst, bDst;
 
-		wm->decomposeColor(src, rSrc, gSrc, bSrc);
-		wm->decomposeColor(*dst, rDst, gDst, bDst);
+		wm->decomposeColor<T *>(src, rSrc, gSrc, bSrc);
```

Comment:
Since you're using `<T *>` everywhere, I suggest changing the template to `<uint32>` and `<byte>`, so you could drop the star.